src/pages/Post.jsx

```diff
@@ -1,6 +1,114 @@
-import { useParams } from "react-router-dom";
+import { useParams, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { usePosts } from "../context/PostsContext";
 
 export default function Post() {
   const { id } = useParams();
-  return <h1>Item Detail Page — ID: {id}</h1>;
+  const navigate = useNavigate();
+  const { getPost, currentPost, isLoading, error } = usePosts();
+  const [selectedPhoto, setSelectedPhoto] = useState(null);
+
+  useEffect(() => {
+    getPost(Number(id));
+  }, [id, getPost]);
+
+  useEffect(() => {
+    if (currentPost?.photos?.length > 0) {
+      setSelectedPhoto(currentPost.photos[0]);
+    }
+  }, [currentPost]);
```

Comment:
Alternative implementation for the sake of exercising the solution - this is just a suggestion.

You can drop the second `useEffect` if you use the state to store the index instead of the actual value: 

```suggestion
  const [selectedPhotoIndex, setSelectedPhotoIndex] = useState(0);
  const selectedPhoto = currentPost?.photos?.[selectedPhotoIndex] || null;

  useEffect(() => {
    getPost(Number(id));
  }, [id, getPost]);
```

It is always good to not introduce a new hook if you can. Every hook added has a potential performance impact.